pandas/tseries/tools.py

```diff
@@ -271,6 +285,14 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     >>> pd.to_datetime('13000101', format='%Y%m%d', errors='coerce')
     NaT
     """
+    
+    if origin is not None:
+        from pandas.core.api import Timestamp
+        if origin == 'julian':
+            arg = arg - Timestamp(0).to_julian_date()
+        else:
+            arg = arg + Timestamp(origin).to_julian_date() - Timestamp(0).to_julian_date()
```

Comment:
Since I am subtracting from the raw input, I would have to also account for the default offset behavior. i.e.
```
arg = arg - (defaultOrigin - newOrigin)
```